Formula/metacall.rb

```diff
@@ -0,0 +1,106 @@
+class Metacall < Formula
+  desc "Ultimate polyglot programming experience"
+  homepage "https://metacall.io"
+  url "https://github.com/metacall/core/archive/refs/tags/v0.5.27.tar.gz"
+  sha256 "1493afadac893b7b3674ea44dd7caa7fd74989880658ae23003b30d3721bb92d"
+  license "Apache-2.0"
+  head "https://github.com/metacall/core.git", branch: "develop"
+
+  depends_on "cmake" => :build
+  depends_on "node@14"
+  depends_on "openjdk"
+  depends_on "python@3.9"
+  uses_from_macos "ruby"
+
+  def install
+    Dir.mkdir("build")
+    Dir.chdir("build")
+    args = std_cmake_args + %W[
+      -Wno-dev
+      -DCMAKE_BUILD_TYPE=Release
+      -DOPTION_BUILD_SECURITY=OFF
+      -DOPTION_FORK_SAFE=OFF
+      -DOPTION_BUILD_SCRIPTS=OFF
+      -DOPTION_BUILD_TESTS=OFF
+      -DOPTION_BUILD_EXAMPLES=OFF
+      -DOPTION_BUILD_LOADERS_PY=ON
+      -DOPTION_BUILD_LOADERS_NODE=ON
+      -DNodeJS_INSTALL_PREFIX=/usr/local/Cellar/metacall/#{version}
+      -DOPTION_BUILD_LOADERS_JAVA=ON
+      -DOPTION_BUILD_LOADERS_JS=OFF
+      -DOPTION_BUILD_LOADERS_C=OFF
+      -DOPTION_BUILD_LOADERS_COB=OFF
+      -DOPTION_BUILD_LOADERS_CS=OFF
+      -DOPTION_BUILD_LOADERS_RB=ON
+      -DOPTION_BUILD_LOADERS_TS=ON
+      -DOPTION_BUILD_LOADERS_FILE=ON
+      -DOPTION_BUILD_PORTS=ON
+      -DOPTION_BUILD_PORTS_PY=ON
+      -DOPTION_BUILD_PORTS_NODE=ON
+    ]
+    system "cmake", *args, ".."
+    system "cmake", "--build", ".", "--target", "install"
+
+    shebang = "\#!/usr/bin/env bash\n"
+    # debug = "set -euxo pipefail\n"
+
+    metacall_extra = [
+      "LOC=/usr/local/Cellar/metacall/#{version}\n",
+      "export LOADER_LIBRARY=\"$LOC/lib\"\n",
+      "export SERIAL_LIBRARY_PATH=\"$LOC/lib\"\n",
+      "export DETOUR_LIBRARY_PATH=\"$LOC/lib\"\n",
+      "export PORT_LIBRARY_PATH=\"$LOC/lib\"\n",
+      "export CONFIGURATION_PATH=\"$LOC/configurations/global.json\"\n",
+    ]
```

Comment:
This should be patched upstream to be configured by cmake.

Reply:
I will work upstream so that this be configured by CMake if possible, thanks for your feedback

Reply:
@SMillerDev I will let someone upstream, answer your question about libexec. However, another failed step in the CI/CD is the tests, do you know of any variable that Homebrew defines the installation folder ? The usual path is `/usr/local/Cellar/metacall/#{version}`.

To sum up, as of now, upstream has advised to wrap executing the binary inside a Bash script to be able to define ENV variables for the binary and a `LOC` variable pointing to the MetaCall installation folder, however in the CI/CD the folder is different `/opt/homebrew/Cellar/...`.

Any, recommendation on how to fix this ?

![image](https://user-images.githubusercontent.com/14914796/187087695-32e29d96-c6ac-42bb-909c-12a6aa3797a3.png)

Reply:
https://docs.brew.sh/Formula-Cookbook#variables-for-directory-locations

Reply:
Hi, I am the author of MetaCall/Core, CMake does not provide support for installing environment variables (https://cmake.org/cmake/help/latest/command/install.html). In addition, this launcher needs to setup some extra behavior that has not been implemented yet in the brew formulae which is usually platform or package manager dependent.

This project is really hard to package, that's why we do not provide such things installed directly from upstream. As you can see we depend on multiple language runtimes that do not have a common design and it is hard to manage.